### PR TITLE
added hue shift image node

### PR DESCRIPTION
**Pull request by MrSimmmons**

### CODE REVIEW

File: app/script/nodes/Image/Effects/NodeHueShift.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+		this.matrix[2][2] = cosA + (1 / 3) * (1.0 - cosA);
+	}
+
+	private radians(degrees: number): number {
+		return (degrees * Math.PI) / 180;
+	}
+
+	private pixelClamp(value: number): number {
+		return Math.max(0, Math.min(255, value));
+	}
```
Comment (oparisblue) [inline app/script/nodes/Image/Effects/NodeHueShift.ts]: Maybe these should be extracted out into util files?

File: app/script/nodes/Image/Effects/NodeHueShift.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
+///<reference path="../NodeImageTransformation.ts" />
+
+/**
+ * @classdesc "Hue" shift the colours on an image by a given amount
+ * @author Simon, Orlando
+ */
+
+@register
+class NodeHueShift extends NodeImageTransformation {
```
Comment (oparisblue) [inline app/script/nodes/Image/Effects/NodeHueShift.ts]: Will need to export now, add the relevant imports, and remove the `///reference`

File: app/script/nodes/Image/Effects/NodeHueShift.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+			pixels[i] = rx;
+			pixels[i + 1] = gx;
+			pixels[i + 2] = bx;
+		}
+
+		// Apply the transformations to the canvas
+		ctx.putImageData(data, 0, 0);
+	};
+
+	private set_hue_rotation(degrees: number) {
```
Comment (oparisblue) [inline app/script/nodes/Image/Effects/NodeHueShift.ts]: Most other functions in the app are in camel case, we should probably be consistent

File: app/script/nodes/Image/Effects/NodeHueShift.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+		this.matrix[0][0] = cosA + (1.0 - cosA) / 3.0;
+		this.matrix[0][1] = (1 / 3) * (1.0 - cosA) - Math.sqrt(1 / 3) * sinA;
+		this.matrix[0][2] = (1 / 3) * (1.0 - cosA) + Math.sqrt(1 / 3) * sinA;
+		this.matrix[1][0] = (1 / 3) * (1.0 - cosA) + Math.sqrt(1 / 3) * sinA;
+		this.matrix[1][1] = cosA + (1 / 3) * (1.0 - cosA);
+		this.matrix[1][2] = (1 / 3) * (1.0 - cosA) - Math.sqrt(1 / 3) * sinA;
+		this.matrix[2][0] = (1 / 3) * (1.0 - cosA) - Math.sqrt(1 / 3) * sinA;
+		this.matrix[2][1] = (1 / 3) * (1.0 - cosA) + Math.sqrt(1 / 3) * sinA;
+		this.matrix[2][2] = cosA + (1 / 3) * (1.0 - cosA);
```
Comment (oparisblue) [inline app/script/nodes/Image/Effects/NodeHueShift.ts]: Any reason some of these are `1.0`, `3.0` etc and some are `1`, `2`, etc? Can remove the decimals on all of them?

File: app/script/nodes/Image/Effects/NodeHueShift.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+ * @classdesc "Hue" shift the colours on an image by a given amount
+ * @author Simon, Orlando
+ */
+
+@register
+class NodeHueShift extends NodeImageTransformation {
+	private matrix: number[][];
+
+	constructor() {
+		super("Hue Shift", "Shift the colours of an image", "/Effects");
```
Comment (oparisblue) [inline app/script/nodes/Image/Effects/NodeHueShift.ts]: Can we call this node / class "Hue Rotate" actually? I think that's typically what it's called in e.g. Photoshop

**Review by oparisblue — COMMENTED** [review]
Overall looks good, just a few minor comments

File: app/script/nodes/Image/Effects/NodeHueShift.ts (hunk 4 above)
Comment (MrSimmmons) [reply app/script/nodes/Image/Effects/NodeHueShift.ts]: The source (https://stackoverflow.com/questions/8507885/shift-hue-of-an-rgb-color) was written in C so I'm guessing they are wanted to be floats there. 
Classic copy paste ;)

**Review by oparisblue — COMMENTED** [review]
Looks good, just need to rename `NodeHueShift.ts` to `NodeHueRotate.ts`

**Review by oparisblue — APPROVED** [review]
lgtm

**Pull request merged**